pkg/services/sqlstore/migrations/datasource_mig.go

```diff
@@ -133,4 +133,22 @@ func addDataSourceMigration(mg *Migrator) {
 
 	const setEmptyJSONWhereNullJSON = `UPDATE data_source SET json_data = '{}' WHERE json_data is null`
 	mg.AddMigration("Update json_data with nulls", NewRawSqlMigration(setEmptyJSONWhereNullJSON))
+
+	// add column Uid column for linking
+	mg.AddMigration("Add uid column", NewAddColumnMigration(tableV2, &Column{
+		Name: "uid", Type: DB_Text, Nullable: false, Default: "0",
```

Comment:
Same as dashboard migration.

```suggestion
		Name: "uid", Type: DB_NVarchar, Length: 40, Nullable: true,
```

Reply:
btw what is the reason for that column type? Seems like nullability actually should not be allowed and not sure what is the benefit in DB_NVarchar over Text. Can't be length enforced with type Text?

Reply:
> Can't be length enforced with type Text?

Yes exactly

Reply:
And regarding the nullability? I am not sure about Dashboards but here it could prevent an issue as not having Uid would break the linking in derived fields config for example.

Reply:
You cannot set it to non-nullable because migration will fail when there are existing rows in data_source table.

Reply:
`uid` not being null for dashboard is prevented in the codebase. Same could be expected for datasources. Nullable false would be preferred but it will require extra steps in the migrations.

1. Add nullable column
2. Make sure all columns have value
3. Make column not nullable

I think nullable fields work with a uniqueness index as well.  https://github.com/grafana/grafana/blob/master/pkg/services/sqlstore/migrations/dashboard_mig.go#L163

Reply:
Well it works right now even with existing rows, that's why it has `Default: "0"`. That needs to be changed later for the uniqueness but we add the unique constraint later after we fill it with unique values.

Reply:
I think you'll see that integration tests may fail when you've resolved existing CI failure

Reply:
Think only one row per ord id and null uid is allowed for the unique index. Uid should always be set AFAIK

Reply:
Seems to work fine as it is right now `Nullable: false, Default: "0"`